### PR TITLE
Fix geospacial queries to use the MongoDB index

### LINKED ISSUE #170: Bad API Query Performance (1000ms for a basic bounding box request)
I have noticed a quite severe performance degradation of the current OCM public API. For example:

```
export ocm_key="some-valid-api-key-here"

time curl -H "X-API-Key: $ocm_key" -s https://api.openchargemap.io/v3/poi'?output=json&maxresults=1&compact=true&verbose=false&camelcase=true&boundingbox=(48.8,9.1),(48.7,9.2)' >/dev/null

curl -H "X-API-Key: $ocm_key" -s  > /dev/null  0,02s user 0,01s system 2% cpu 1,094 total
```

This very basic request, just using a bounding box for filtering and limiting the result set to one single record takes about one second to complete.

I did also setup a OCM Mirror as described [here](https://github.com/openchargemap/ocm-system/issues/148) to see if the issue persists and also to be able to debug it. And, yes, it does:

```
time curl -H "X-API-Key: $ocm_key" -s https://ocm-mirror.io.ev-freaks.com/v3/poi'?output=json&maxresults=1&compact=true&verbose=false&camelcase=true&boundingbox=(48.8,9.1),(48.7,9.2)' >/dev/null
curl -H "X-API-Key: $ocm_key" -s  > /dev/null  0,01s user 0,00s system 3% cpu 0,480 total
```

I am getting better results using my own mirror (480ms vs. 1094ms) but this is still quite bad..

I was also looking at the mongoDB requests in docker and I saw this kind of queries:

```
(..)
		"command" : {
			"$truncated" : "{ find: \"poi\", filter: { $and: [ { SpatialPosition: { $within: { $polygon: [ [ 9.077199999999999, 48.8741 ], [ 9.239000000000001, 48.8741 ], [ 9.239000000000001, 49.0802 ], [ 9.077199999999999, 49.0802 ], [ 9.077199999999999, 48.8741 ] ] } } }, { AddressInfo: { $ne: null } }, { $or: [ { SubmissionStatusTypeID: null }, { SubmissionStatusTypeID: 100 }, { SubmissionStatusTypeID: 200 }, { _id: { $type: -1 } }, { $and: [ { SubmissionStatusTypeID: { $ne: null } }, { _id: { $type: -1 } }, { SubmissionStatusTypeID: null } ] } ] }, { SubmissionStatusTypeID: { $ne: null } }, { SubmissionStatusTypeID: { $ne: 1010 } }, { $or: [ { StatusTypeID: { $ne: 200 } }, { StatusTypeID: { $in: [ -1 ] }, _id: { $type: -1 } } ] }, { Connections: { $elemMatch: { $or: [ { ConnectionType.ID: { $in: [ 25, 1036, 33, 28 ] } }, { _id: { $type: -1 } } ] } } } ] }, sort: { ID: -1 }, skip: 0, limit: 501, noCursorTimeout: false, $db: \"ocm_mirror\", lsid: { id: UUID(\"7e18d1b8-9720-4a14-8ce2-274315bed0bd\") } }"
		},
(..)
```

IMO `$within` **will not use** the `2dsphere` index. To leverage the `2dsphere` index the `$geoWithin` operator has to be used, see also

https://docs.mongodb.com/manual/reference/operator/query/geoWithin/

For my projects I was able to get a way better query performance using the "2d" index (instead of the 2dsphere), btw.

Unfortunately I am a Swift Developer and I do not have any .NET expertise :(

### PR DESCRIPTION
**Pull request by remuslazar**
Resolves #170 

## TOC

1. Use the geospatial "2d" index for boundingbox queries
2. Do not sort the result set when doing boundingbox queries for performance reasons

## TL;DR

This changes the existing MongoDB geospatial index in "2d" and rework the query logic to use that.

Prior to this PR MongoDB was not able to use the geospatial index when doing boundingbox queries and this lead to a major performance degradation and high RAM usage on the MongoDB server instance. See #170 for details.

### Technical Stuff

MongoDB uses the `2dsphere` index only for queries with the `$geometry` operator, see https://docs.mongodb.com/manual/tutorial/query-a-2dsphere-index/. For "basic" `$polygon` queries only the legacy `2d` index can be used.

To leverage the `2dsphere` index a newer version of the MongoDB Library has to be used. The legacy version of the MongoDB driver (1.x) (currently used throughout the codebase) can not do that, unfortunately.

### Benchmarks

On my local machine the query time for a boundingbox query went down from ~400ms to ~30ms. Using the index the mongo instance runs also fine with 512MB RAM in docker.

### Sorting Behavior Change

Additionally this change also skips the sorting of the result set by ID when doing boundingbox queries to get even more performance. This is especially a thing when dealing with large result sets, e.v. 500 records or more.

### CODE REVIEW

**Comment by webprofusion-chrisc** [comment]
Thanks, this is awesome! Regarding the removal of the sort, we try to keep the results set the same between the SQL implementation and the MongoDB implementation so that we can automate a few tests but I'll merge this and see how I get on.
